pkg/controller/node/nodecontroller.go

```diff
@@ -740,35 +740,37 @@ func (nc *NodeController) tryUpdateNodeStatus(node *v1.Node) (time.Duration, v1.
+				currentReadyCondition.Message = "Kubelet stopped posting node status."
 				// LastProbeTime is the last time we heard from kubelet.
 				currentReadyCondition.LastHeartbeatTime = observedReadyCondition.LastHeartbeatTime
 				currentReadyCondition.LastTransitionTime = nc.now()
 			}
 		}
 
-		// Like NodeReady condition, NodeOutOfDisk was last set longer ago than gracePeriod, so update
-		// it to Unknown (regardless of its current value) in the master.
-		// TODO(madhusudancs): Refactor this with readyCondition to remove duplicated code.
-		_, oodCondition := v1.GetNodeCondition(&node.Status, v1.NodeOutOfDisk)
-		if oodCondition == nil {
-			glog.V(2).Infof("Out of disk condition of node %v is never updated by kubelet", node.Name)
-			node.Status.Conditions = append(node.Status.Conditions, v1.NodeCondition{
-				Type:               v1.NodeOutOfDisk,
-				Status:             v1.ConditionUnknown,
-				Reason:             "NodeStatusNeverUpdated",
-				Message:            fmt.Sprintf("Kubelet never posted node status."),
-				LastHeartbeatTime:  node.CreationTimestamp,
-				LastTransitionTime: nc.now(),
-			})
-		} else {
-			glog.V(4).Infof("node %v hasn't been updated for %+v. Last out of disk condition is: %+v",
-				node.Name, nc.now().Time.Sub(savedNodeStatus.probeTimestamp.Time), oodCondition)
-			if oodCondition.Status != v1.ConditionUnknown {
-				oodCondition.Status = v1.ConditionUnknown
-				oodCondition.Reason = "NodeStatusUnknown"
-				oodCondition.Message = fmt.Sprintf("Kubelet stopped posting node status.")
-				oodCondition.LastTransitionTime = nc.now()
+		// remaining node conditions should also be set to Unknown
+		remainingNodeConditionTypes := []v1.NodeConditionType{v1.NodeOutOfDisk, v1.NodeMemoryPressure, v1.NodeDiskPressure, v1.NodeNetworkUnavailable}
+
+		for _, nodeConditionType := range remainingNodeConditionTypes {
+			_, currentCondition := v1.GetNodeCondition(&node.Status, nodeConditionType)
+			if currentCondition == nil {
+				glog.V(2).Infof("Condition %v of node %v was never updated by kubelet", nodeConditionType, node.Name)
+				node.Status.Conditions = append(node.Status.Conditions, v1.NodeCondition{
+					Type:               nodeConditionType,
+					Status:             v1.ConditionUnknown,
+					Reason:             "NodeStatusNeverUpdated",
+					Message:            "Kubelet never posted node status.",
+					LastHeartbeatTime:  node.CreationTimestamp,
+					LastTransitionTime: nc.now(),
+				})
+			} else {
+				glog.V(4).Infof("node %v hasn't been updated for %+v. Last %v is: %+v",
+					node.Name, nc.now().Time.Sub(savedNodeStatus.probeTimestamp.Time), nodeConditionType, currentCondition)
+				if currentCondition.Status != v1.ConditionUnknown {
+					currentCondition.Status = v1.ConditionUnknown
+					currentCondition.Reason = "NodeStatusUnknown"
+					currentCondition.Message = "Kubelet stopped posting node status."
+					currentCondition.LastTransitionTime = nc.now()
```

Comment:
We probably want to have the same timestamp for all condition, so please grab 'now' before the loop.